src/cursor/abstract_cursor.ts

```diff
@@ -284,7 +289,7 @@ export abstract class AbstractCursor<
   next<T = TSchema>(callback?: Callback<T | null>): Promise<T | null> | void {
     return maybePromise(callback, done => {
       if (this[kId] === Long.ZERO) {
-        return done(new MongoDriverError('Cursor is exhausted'));
+        return done(new MongoCursorExhaustedError('Cursor is exhausted'));
```

Comment:
Have we considered moving this generic message to the error constructor? Do we think there might be a different message related to the same error?

Reply:
That's a really good point. For many of these errors they really only need one default message as far as I can see.

Do you think it would be a good idea to change those specific-use errors to have a default generic message, but still accept an optional message argument in case it may prove useful at some point in the future?

Reply:
Yes, I think that would be a good approach, to allow a message override, though I don't know if you need to even include it right now, it would be just as easy to add when we have an actual use case for it